tests/unit/externalRequestsTest.js

```diff
@@ -81,13 +294,172 @@ tap.test('External Requests Middleware', {autoend: true}, t => {
           'prepareLookupRequests'
         )
 
-        prepareLookupRequests(ctx).then(() => {
-          t.same({test1: 'testA', test2: 'testB'}, ctx.lookupRequests)
+        prepareLookupRequests(ctx).catch(err => {
           performRequestsStub()
+          t.type(err, Error)
+          t.equal(err.message, 'Rejected Promise: Fail')
         })
-      }
-    )
+      })
+
+      t.test(
+        'should add response data to the ctx when all promises resolve',
+        t => {
+          t.plan(1)
 
+          const performRequestsStub = externalRequests.__set__(
+            'performRequests',
+            () => [
+              Promise.resolve({test1: 'testA'}),
+              Promise.resolve({test2: 'testB'})
+            ]
+          )
+
+          const ctx = {
+            state: {
+              uuid: 'randomUidForRequest',
+              metaData: {
+                name: 'Testing endpoint',
+                requests: {
+                  lookup: [
+                    {
+                      // first lookup request config - responds with success
+                    },
+                    {
+                      // second lookup request config - responds with success
+                    }
+                  ]
+                }
+              }
+            }
+          }
+
+          const prepareLookupRequests = externalRequests.__get__(
+            'prepareLookupRequests'
+          )
+
+          prepareLookupRequests(ctx).then(() => {
+            t.same({test1: 'testA', test2: 'testB'}, ctx.lookupRequests)
+            performRequestsStub()
+          })
+        }
+      )
+      t.test(
```

Comment:
add a new line above this test declaration

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/39/commits/f6e854686999d065f57f20f9f1af3f9e1318f21e